charts/opentelemetry-operator/templates/deployment.yaml

```
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: {{ template "opentelemetry-operator.name" . }}-controller-manager
+  namespace: {{ template "opentelemetry-operator.namespace" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  template:
+    metadata:
+      labels:
+        control-plane: controller-manager
+    spec:
+      containers:
+        - args:
+            - --metrics-addr=127.0.0.1:8080
+            - --enable-leader-election
+          command:
+            - /manager
+          image: "{{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag }}"
+          name: manager
+          ports:
+            - containerPort: 9443
+              name: webhook-server
+              protocol: TCP
+          resources:
+{{ toYaml .Values.manager.resources | indent 12 }}
+          volumeMounts:
+            - mountPath: /tmp/k8s-webhook-server/serving-certs
+              name: cert
+              readOnly: true
+        - args:
+            - --secure-listen-address=0.0.0.0:8443
+            - --upstream=http://127.0.0.1:8080/
+            - --logtostderr=true
+            - --v=10
+          image: "{{ .Values.kubeRBACProxy.image.repository }}:{{ .Values.kubeRBACProxy.image.tag }}"
+          name: kube-rbac-proxy
+          ports:
+            - containerPort: 8443
+              name: https
+      terminationGracePeriodSeconds: 10
+      volumes:
+        - name: cert
+          secret:
+            defaultMode: 420
```

```suggestion
            defaultMode: 400
```

Don't think group write mode makes sense for a read-only volume that the group can't read.

---

I agree. This makes sense. I inherited this from the upstream version of this deployment resource. Maybe there is a mistake.

---

It turns out if we change the mode to 400, which removes the write permission for group users, the following error would occur when installing the Operator
```console
{"level":"error","ts":1624400990.3124812,"logger":"setup",
"msg":"problem running manager",
"error":"open /tmp/k8s-webhook-server/serving-certs/tls.crt: permission denied",
"stacktrace":"main.main\n\t/workspace/main.go:172\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204"}
```
Seems like we need to give the write permission to group users.